proxy/http2/HTTP2.cc

```diff
@@ -406,88 +406,76 @@ http2_parse_window_update(IOVec iov, uint32_t &size)
 ParseResult
 http2_convert_header_from_2_to_1_1(HTTPHdr *headers)
 {
-  MIMEField *field;
-
   ink_assert(http_hdr_type_get(headers->m_http) != HTTP_TYPE_UNKNOWN);
 
+  // HTTP Version
+  headers->version_set(HTTPVersion(1, 1));
+
   if (http_hdr_type_get(headers->m_http) == HTTP_TYPE_REQUEST) {
-    const char *scheme, *authority, *path;
-    int scheme_len, authority_len, path_len;
+    // :scheme
+    if (MIMEField *field = headers->field_find(HTTP2_VALUE_SCHEME, HTTP2_LEN_SCHEME); field != nullptr && field->value_is_valid()) {
+      int scheme_len;
+      const char *scheme = field->value_get(&scheme_len);
+
+      int scheme_wks_idx = hdrtoken_tokenize(scheme, scheme_len);
+      url_scheme_set(headers->m_heap, headers->m_http->u.req.m_url_impl, scheme, scheme_wks_idx, scheme_len, true);
```

Comment:
Does this work even if the scheme is not wks? Or can we ignore the case?

Reply:
IIUC, `url_scheme_set()` copy `scheme` when `bool copy_string` flag (the last arg) is `true` even if `scheme_wks_idx` is `-1`.